include/kllvm/ast/AST.h

```diff
@@ -963,7 +966,17 @@ class kore_definition {
    *
    *   S |-> {T . S is a subsort of T}
    */
-  [[nodiscard]] SubsortMap get_subsorts() const;
+  [[nodiscard]] SubsortMap get_subsorts();
+
+  /*
+   * Build this definition's inverted subsort relation from axioms that have the
+   * `subsort` attribute.
+   *
+   * The returned map is as follows:
+   *
+   * S |-> {T . S is a direct supersort of T}
```

Comment:
This isn't the case in your implementation; my preference is to just cache the transitive closure result for consistency.